pkg/network/vlan/vlan.go

```diff
@@ -114,58 +119,58 @@ func (v *Vlan) Teardown() error {
 	return nil
 }
 
-func (v *Vlan) AddLocalArea(vid uint16, cidr string) error {
+func (v *Vlan) AddLocalArea(la *LocalArea) error {
 	if v.uplink == nil {
 		return fmt.Errorf("bridge %s hasn't attached an uplink", v.bridge.Name)
 	}
-	if ok, _ := v.findVid(vid); ok {
+	if ok, _ := v.findVid(la.Vid); ok {
 		return nil
 	}
 
-	if err := v.uplink.AddBridgeVlan(vid); err != nil {
-		return fmt.Errorf("add bridge vlanconfig %d failed, error: %w", vid, err)
+	if err := v.uplink.AddBridgeVlan(la.Vid); err != nil {
+		return fmt.Errorf("add bridge vlanconfig %d failed, error: %w", la.Vid, err)
 	}
-	v.vids = append(v.vids, vid)
+	v.localAreas = append(v.localAreas, la)
```

Comment:
put `append vid` behind `iface.EnsureRouteViaGateway`, for reconciler to proper retry

question: the return result of `v.uplink.AddBridgeVlan(vid)` when vid is existing?  considering reconciler's retry

Reply:
It won't return any error if the vid is existing.